src/components/tl/mlx5/mcast/tl_mlx5_mcast_coll.c

```diff
@@ -185,34 +443,45 @@ static inline ucc_status_t ucc_tl_mlx5_mcast_prepare_bcast(void* buf, size_t siz
     req->proto   = (req->length < comm->max_eager && !comm->cuda_mem_enabled) ?
                             MCAST_PROTO_EAGER : MCAST_PROTO_ZCOPY;
 
-    status = ucc_tl_mlx5_mcast_prepare_reliable(comm, req, req->root);
-    if (ucc_unlikely(UCC_OK != status)) {
-        return status;
+    req->num_packets = ucc_max(ucc_div_round_up(req->length, comm->max_per_packet), 1);
```

Comment:
taking the max should not be necessary, right?

Reply:
yes, I fixed it, thanks!